src/app/components/FileBox.tsx

```diff
@@ -1,45 +1,81 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-import { Body, Box } from '@app/components';
+import { Body, Box, BoxProps, Image } from '@app/components';
+import checkmark from '@assets/icons/circle-checkmark.svg';
 import { translateRaw } from '@translations';
 
-export const FileBox = ({ onChange }: { onChange: React.ChangeEventHandler<HTMLInputElement> }) => (
-  <Box
-    bg="BG_GREY_MUTED"
-    p="2"
-    sx={{
-      borderWidth: '1px',
-      borderStyle: 'solid',
-      borderColor: 'GREY_ATHENS',
-      boxShadow: '0px 1px 1px rgba(232, 234, 237, 0.5), inset 0px 1px 3px rgba(232, 234, 237, 0.5)',
-      borderRadius: '6px'
-    }}
-    // @todo Fix OnDrop
-    //onDrop={() => console.log('drop')}
-  >
+export const FileBox = ({
+  onChange,
+  ...props
+}: { onChange(file: File): void } & Omit<BoxProps, 'onChange'>) => {
+  const [fileName, setFileName] = useState<string | undefined>(undefined);
+
+  // Needs to be set for onDrop to work
+  const handleDragOver = (e: React.DragEvent) => {
```

Comment:
Something to consider: The new JSX transform will remove the need for `React` imports. It may be more future-proof to import directly, e.g. `import { DragEvent } from 'react';`, since then we can easily remove all `React` imports if we deccide to switch to the JSX transform. Then the imports from React are essentially type-only imports.